### DISCUSSION TITLE
Proposal for new testing procedure

### DISCUSSION THREAD

**Discussion by adkinsrs** [IGS/gEAR#117] [Ideas]
Ronna and I were talking today about how to streamline testing for new features so that the test users will just do the tests without having to think about testing.

1. Create issue on Github
2. When new feature or bugfix is complete, create a new issue to test the previously created issue.
3. In the "test" issue we:
   * Assign the task to the test team
   * Cross-reference the original issue we are testing
   * Give directions on how to perform the new issue (if applicable).
   * Test on the devel server.
4. When testing is complete, the test issue is closed, which signs off on closing the original issue 

@jorvis, thoughts on this?

**Comment by jorvis** [comment]
I like this, and I believe GitHub has templating systems that can be used here to make this easier for when there's a testing ticket.

**Comment by adkinsrs** [comment]
What I am implementing:

1. Create issue on Github
2. When new feature or bugfix is complete, add a "needs testing" tag, and tag relevant users.
3. Give directions to reproduce or perform task if applicable.
4. Have the "back and forth" with them on if it works or not... rinse... repeat.
5. When the test crew gives the OK, then we mark the test as complete.

**Comment by jorvis** [comment]
Thoughts on the tests themselves being separate tickets?  First, the testers may not need to see all the discussion, etc. that went toward creating it, and I know there's some developer satisfaction in being able to close a ticket after they have written and tested something.  They could then create the testing ticket and reference the initial one if appropriate.

**Reply by adkinsrs** [reply]
If the tester finds something wrong, does the developer reopen the original ticket, or does the developer put fixes in the "test" ticket?  Since Github also keeps track of chronological actions, the tester could theoretically ignore anything above the "testing" label in the original ticket.  I certainly don't have a problem with creating a separate ticket, but I'm just throwing out scenarios.

**Reply by jorvis** [reply]
Let's talk about this in the update meeting tomorrow and get people's thoughts.  There are guides (which we can, of course, disagree with):

https://www.thirdandgrove.com/insights/tips-on-testing-in-an-agile-environment-and-pitfalls-to-avoid/

**Comment by adkinsrs** [comment]
When I create a "needs testing" ticket, who should be the default assignees?  I would propose @JPReceveur , @songeric1107 , @Kvnrose , and maybe Brian Herb (who is not in the IGS Github organization).

**Reply by jorvis** [reply]
Writing Victor now to get Brian added into the IGS org.
